store/src/directory.rs

```diff
@@ -392,21 +812,80 @@ where
                 Ok(v) if v.is_file() => {}
                 Ok(_) => continue,
             }
-            let ttl = match xattr::get(&path, format_xattr(crate::MetadataKey::<MKT>::Ttl.to_key()))
-            {
-                Err(e) => {
-                    log::trace!("Error looking up TTL xattr for {}: {:?}", path.display(), e);
+            // skip non-metadata files if xattr is not supported
+            if !self.xattr_enabled {
+                match path.as_path().extension() {
+                    None => continue,
+                    Some(ex) => {
+                        if ex != FDO_METADATA_EX {
+                            continue;
+                        }
+                    }
+                }
+                let path_meta = set_metadata_extension_to_path(&path)?;
```

Comment:
uhm, if the paths here are _only_ metadata files, why doing this? and also, when removing the ov and metadata below, how do you actually remove the OV if we get here only if it's metadata file